### PR TITLE
[src & egs] Prepare lightning upgrade to 0.9.0

### PR DESCRIPTION
**Pull request by mpariente**
- `optimizer_step` change
- Fix trainer args
- Fix verbose is boolean

### CODE REVIEW

**Comment by mpariente** [comment]
@popcornell, does this solve #213 by change?

File: requirements.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@ pandas>=0.23.4
 scipy>=1.1.0
 SoundFile>=0.10.2
 torch>=1.3.0
-pytorch-lightning>=0.7.5,<0.8
+pytorch-lightning>=0.8
```
Comment (popcornell) [inline requirements.txt:7]: wouldn't be better to use == because new lightning releases tend to break things

**Comment by popcornell** [comment]
> @popcornell, does this solve #213 by chance?

I need to wait for a free GPU to test it

File: requirements.txt (hunk 1 above)
Comment (mpariente) [reply requirements.txt:7]: We'll see, release 0.9.0 will be released soon and we can test against it. 
If things break, we'll upgrade to 0.9.0 directly

**Comment by popcornell** [comment]
Nope still not solved. 

I am getting this rather un-helpful error: 

```
File "train.py", line 140, in <module>
    main(arg_dic)
  File "train.py", line 104, in main
    trainer.fit(system)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/pytorch_lightning/trainer/trainer.py", line 992, in fit
    results = self.spawn_ddp_children(model)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/pytorch_lightning/trainer/distrib_data_parallel.py", line 462, in spawn_ddp_children
    results = self.ddp_train(local_rank, q=None, model=model, is_master=True)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/pytorch_lightning/trainer/distrib_data_parallel.py", line 560, in ddp_train
    results = self.run_pretrain_routine(model)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/pytorch_lightning/trainer/trainer.py", line 1196, in run_pretrain_routine
    False)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/pytorch_lightning/trainer/evaluation_loop.py", line 291, in _evaluate
    output = self.evaluation_forward(model, batch, batch_idx, dataloader_idx, test_mode)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/pytorch_lightning/trainer/evaluation_loop.py", line 444, in evaluation_forward
    output = model(*args)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/pytorch_lightning/overrides/data_parallel.py", line 96, in forward
    output = self.module.validation_step(*inputs[0], **kwargs[0])
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/asteroid-0.3.0-py3.6.egg/asteroid/engine/system.py", line 136, in validation_step
    loss = self.common_step(batch, batch_nb, train=False)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/asteroid-0.3.0-py3.6.egg/asteroid/engine/system.py", line 91, in common_step
    est_targets = self(inputs)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/asteroid-0.3.0-py3.6.egg/asteroid/engine/system.py", line 64, in forward
    return self.model(*args, **kwargs)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/asteroid-0.3.0-py3.6.egg/asteroid/models/base_models.py", line 49, in forward
    est_masks = self.masker(tf_rep)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/asteroid-0.3.0-py3.6.egg/asteroid/masknn/recurrent.py", line 352, in forward
    output = self.net(output)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/torch/nn/modules/container.py", line 117, in forward
    input = module(input)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/asteroid-0.3.0-py3.6.egg/asteroid/masknn/recurrent.py", line 212, in forward
    x = self.intra_RNN(x)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/asteroid-0.3.0-py3.6.egg/asteroid/masknn/recurrent.py", line 50, in forward
    rnn_output, _ = self.rnn(output)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/scornell/anaconda3/envs/plup/lib/python3.6/site-packages/torch/nn/modules/rnn.py", line 577, in forward
    self.dropout, self.training, self.bidirectional, self.batch_first)
RuntimeError: cuDNN error: CUDNN_STATUS_BAD_PARAM
```

**Comment by popcornell** [comment]
Interestingly this does not anymore depend on the overriding of `optimizer.step()`. It may be a totally unrelated issue.

**Comment by jonashaag** [comment]
FP16 LSTM is broken in PyTorch 1.6, see https://github.com/pytorch/pytorch/issues/36428

I tried FP16 with PyTorch nightly, which doesn't error, but training did not converge in my experiments with DPRNN and DCUNet (not part of Asteroid)

File: asteroid/engine/system.py [hunk 2]
```diff
@@ -114,18 +114,11 @@ def training_step(self, batch, batch_nb):
...
-    def optimizer_step(
-        self,
-        epoch: int,
-        batch_idx: int,
-        optimizer: Optimizer,
-        optimizer_idx: int,
-        second_order_closure: Optional[Callable] = None,
-    ) -> None:
+    def optimizer_step(self, *args, **kwargs):
         for sched in self.scheduler:
```
Comment (jonashaag) [inline asteroid/engine/system.py:118]: If think this misses a `if self.scheduler is not None:`

Comment (mpariente) [reply asteroid/engine/system.py:118]: You're right, thanks

Comment (mpariente) [reply asteroid/engine/system.py:118]: It's also missing in the current code then, which was just released haha. 
I'll see if it breaks (probably) and release 0.3.2 then.

**Comment by mpariente** [comment]
I ran the tests locally and the ConvTasNet recipe (from stage 3) with the following versions : 
- [x] 0.8.1
- [x] 0.8.3 
- [x] 0.8.4
- [x] 0.8.5
- [x] 0.9.0

We're merging this tomorrow unless any of you has something against it @popcornell @jonashaag @faroit @michelolzam ?

**Pull request merged**